src/Akeneo/Bundle/ElasticSearchBundle/Client.php

```diff
@@ -0,0 +1,140 @@
+    private $hosts;
+
+    /** @var string */
+    private $indexName;
+
+    /** @var NativeClient */
+    private $client;
+
+    /**
+     * Configure the PHP ElasticSearch client.
+     * To learn more, please see {@link https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_configuration.html}
+     *
+     * @param ClientBuilder $builder
+     * @param array         $hosts
+     * @param string        $indexName
+     */
+    public function __construct(ClientBuilder $builder, array $hosts, $indexName)
+    {
+        $this->builder = $builder;
+        $this->hosts = $hosts;
+        $this->indexName = $indexName;
+
+        $builder->setHosts($hosts);
+        $this->client = $builder->build();
+    }
+
+    /**
+     * @param string $indexType
+     * @param string $id
+     * @param array  $body
+     *
+     * @return array see {@link https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_quickstart.html#_index_a_document}
+     */
+    public function index($indexType, $id, array $body)
+    {
+        $params = [
+            'index' => $this->indexName,
+            'type'  => $indexType,
+            'id'    => $id,
+            'body'  => $body
+        ];
+
+        return $this->client->index($params);
+    }
+
+    /**
+     * @param string $indexType
+     * @param string $id
+     *
+     * @return array see {@link https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_quickstart.html#_get_a_document}
+     */
+    public function get($indexType, $id)
+    {
+        $params = [
+            'index' => $this->indexName,
+            'type'  => $indexType,
+            'id'    => $id,
+        ];
+
+        return $this->client->get($params);
+    }
+
+    /**
+     * @param string $indexType
+     * @param array  $body
+     *
+     * @return array see {@link https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_quickstart.html#_search_for_a_document}
+     */
+    public function search($indexType, array $body)
+    {
+        $params = [
+            'index' => $this->indexName,
+            'type'  => $indexType,
+            'body'  => $body
+        ];
+
+        return $this->client->search($params);
+    }
+
+    /**
+     * @param string $indexType
+     * @param string $id
+     *
+     * @return array see {@link https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_quickstart.html#_delete_a_document}
+     */
+    public function delete($indexType, $id)
+    {
+        $params = [
+            'index' => $this->indexName,
+            'type'  => $indexType,
+            'id'    => $id
+        ];
+
+        return $this->client->delete($params);
+    }
+
+    /**
+     * @return array see {@link https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_quickstart.html#_delete_an_index}
+     */
+    public function deleteIndex()
+    {
+        return $this->client->indices()->delete(['index' => $this->indexName]);
```

Comment:
Did you check if this throws if the index does not exists ? (otherwise you could add a if ($this->client->indices->hasIndex($indexName)).